File: doc/manuals/processes/sample_process.cxx
#ifndef CLASSIFIER_PROCESS_H
#define CLASSIFIER_PROCESS_H
#include <sprokit/pipeline/process.h>
#include <processes/optical_flow/diva_classifier_process_export.h>
#include <vital/vital_types.h>

namespace diva {

create_config_trait( model_file, vital::path_t, "dummy.model",
                "Model file for the classifier" );
class DIVA_CLASSIFIER_PROCESSES_NO_EXPORT classifier_process
  : public sprokit::process
{
public:
  classifier_process( kwiver::vital::config_block_sptr const& config ) 
      : process( config )
      : d( new classifier_process::priv )
  {
  
    declare_config_using_trait( model_file );
    // Set up flags
    sprokit::process::port_flags_t optional;
    sprokit::process::port_flags_t required;
    required.insert( flag_required );

    // Declare input ports
    declare_input_port_using_trait( image, required );
    declare_input_port_using_trait( file_name, optional );

    declare_output_port_using_trait( double_vector, required );
  }
protected:
  void _configure()
  {
    scoped_configure_instrumentation();
    d->classifer = ClassifierModel(config_value_using_trait( model_file ); 
  }

  void _step()
  {
    auto image_container = grab_from_port_using_trait( image );
    auto file_name = grab_from_port_using_trait( file_name ); 
    std::vector<double> output_classes = d->classifier.classify( image_container->get_image() );
    push_to_port_using_trait( output_classes );
  }

private:
  class priv
  {
    
  };
  const std::unique_ptr<priv> d;
}; // end class classifier_process

}  // end namespace

#endif // CLASSIFIER_PROCESS_H

File: doc/manuals/processes/sample_process.py
from sprokit.pipeline import process
from kwiver.kwiver_process import KwiverProcess

class ClassifierProcess(KwiverProcess):
    def __init__(self, conf):
        KwiverProcess.__init__(self, conf)

        # declare configuration
        self.add_config_trait("model_file", "model_file",
                                'dummy.model', 'Model file for the classifier')
        self.declare_config_using_trait('model_file')

        # set up flags
        required = process.PortFlags()
        required.add(self.flag_required)
        optional = process.PortFlags()
        
        # declare ports
        self.declare_input_port_using_trait('image', required)
        self.declare_input_port_using_trait('file_name', optional )
        self.declare_output_port_using_trait( 'double_vector', required );

    def _configure(self):
        # Configure the process
        self.classifier = Classifier(self.config_value("model_file"))

    def _step(self):
        # Step Function for the process
        img_container = self.grab_input_using_trait('image')
        video_name = self.grab_input_using_trait('file_name')
        # Classify the image
        class_score = self.classifier.classify(img_container.image())
        # Push results to port
        self.push_to_port_using_trait('double_vector', class_score)



def __sprokit_register__():
    from sprokit.pipeline import process_factory
    module_name = 'python:kwiver.ClassifierSample'
    if process_factory.is_process_module_loaded(module_name):
        return
    process_factory.add_process('ClassifierSample', 'Dummy Classifier', 
                                ClassifierProcess)
    process_factory.mark_process_module_as_loaded(module_name)
